File: 0.4._basic_servers/node/main.js
// To make sure node works, run: npm i
// To run this file: node main.js

const express = require('express');
const app = express();
const fs = require('fs');


const folderPath = './me_files/';

// Route for JSON file
app.get('/mejson', (req, res) => {
    const filePath = folderPath + 'me.json';
    const data = JSON.parse(fs.readFileSync(filePath));
    res.json(data);
  });
  
  // Route for CSV file
  app.get('/mecsv', (req, res) => {
    const filePath = folderPath + 'me.csv';
    const data = fs.readFileSync(filePath, 'utf8');
    res.send(data);
  });
  
  // Route for TXT file
  app.get('/metxt', (req, res) => {
    const filePath = folderPath + 'me.txt';
    const data = fs.readFileSync(filePath, 'utf8');
    res.send(data);
  });
  
  // Route for XML file
  app.get('/mexml', (req, res) => {
    const filePath = folderPath + 'me.xml';
    const data = fs.readFileSync(filePath, 'utf8');
    res.send(data);
  });
  
  // Route for YAML file
  app.get('/meyaml', (req, res) => {
    const filePath = folderPath + 'me.yaml';
    const data = fs.readFileSync(filePath, 'utf8');
    res.send(data);
  });

// TEST WITH DUMMY DATA
/* const data = {
  name: 'Leopold',
  age: 28,
  hobbies: ['Music', 'Programming', 'Singing']

  app.get('/me', (req, res) => {
  res.json(data);
});
}; */



app.listen(3000, () => {
  console.log('Server started on port 3000');
});

File: 0.4._basic_servers/python/main.py
# Poetry init, poetry shell, poetry add fastapi uvicorn,
# For at køre serveren skriv i terminalen skriv uvicorn main:app --reload

from fastapi import FastAPI
import json

app = FastAPI()

data_files = ["me.csv", "me.json", "me.txt", "me.xml", "me.yaml"]



@app.get("/")
def root():
    return {"message": "Sup FastAPI nerds"}

@app.get("/newroute")
def _():
    return {"message": "This is a new route"}

# Route to return data from me.json file
@app.get("/mejson")
def get_me_json():
    with open("me_files/me.json", "r") as f:
        data = json.load(f)
    return data

# Route to return data from me.csv file
@app.get("/mecsv")
def get_me_csv():
    with open("me_files/me.csv", "r") as f:
        data = f.read()
    return data

# Route to return data from me.txt file
@app.get("/metxt")
def get_me_txt():
    with open("me_files/me.txt", "r") as f:
        data = f.read()
    return data

# Route to return data from me.xml file
@app.get("/mexml")
def get_me_xml():
    with open("me_files/me.xml", "r") as f:
        data = f.read()
    return data

# Route to return data from me.yaml file
@app.get("/meyaml")
def get_me_yaml():
    with open("me_files/me.yaml", "r") as f:
        data = f.read()
    return data